lib/portage/data.py

```diff
@@ -178,6 +178,17 @@ def _get_global(k):
 				noiselevel=-1)
 			portage_group_warning()
 
+		portage_homedir_writable = os.access(
+				os.path.expanduser("~portage"), portage.os.W_OK)
+		portage_homedir_executable = os.access(
+				os.path.expanduser("~portage"), portage.os.X_OK)
```

Comment:
You can just do a bitwise-OR of all requested flags (and I suppose check `R` then too).

Reply:
Took me a second to realise this meant changing the 2nd `os.access()` param to -

```
portage.os.R_OK | portage.os.W_OK | portage.os.X_OK
```

It's been fixed now though.